File: backend/src/main/java/com/toprakrehberi/backend/dtos/ProductOptionDTO.java
package com.toprakrehberi.backend.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductOptionDTO {

    private Long id;
    private String name;
    private int plantingDuration;
    private String imageUrl;
}

File: frontend/lib/dtos/ProductOptionDTO.dart
class ProductOptionDTO {
  final int id;
  final String name;
  final int plantingDuration;
  final String imageUrl;

  ProductOptionDTO({
    required this.id,
    required this.name,
    required this.plantingDuration,
    required this.imageUrl,
  });

  factory ProductOptionDTO.fromJson(Map<String, dynamic> json) {
    return ProductOptionDTO(
      id: json['id'],
      name: json['name'],
      plantingDuration: json['plantingDuration'],
      imageUrl: json['imageUrl'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'plantingDuration': plantingDuration,
        'imageUrl': imageUrl,
      };
}
